backend/editor/api.py

```diff
@@ -43,14 +43,24 @@ async def startup():
     """
     Initialize database
     """
-    initialize_db()
+    graph_db.initialize_db()
 
 @app.on_event("shutdown")
 async def shutdown():
     """
     Shutdown database
     """
-    shutdown_db()
+    graph_db.shutdown_db()
+
+@app.middleware("http")
+async def initialize_neo4j_transactions(request: Request, call_next):
+    with graph_db.driver.session() as session:
+        with session.begin_transaction() as _txn:
+            # Make it the current transaction
+            graph_db.txn = _txn
+            response = await call_next(request)
+        graph_db.txn = None

```

Comment:
If you're interested we can make this part a context manager in graph_db (so that all code managing db is in there):

```python
@contextlib.contextmanager
def TransactionCtx():
    """Transaction context will set global transaction txn for the code in context

    transaction is automatically rollback if an exception occurs within the context
    """
    global txn
    with driver.session() as session
        with session.begin_transaction() as _txn:
            txn = _txn
            yield txn
    txn = None
```

and here:

```python
@app.middleware("http")
async def initialize_neo4j_transactions(request: Request, call_next):
    with graph_db.TransactionCtx():
        response = await call_next(request)
```